icechunk-python/tests/test_timetravel.py

```diff
@@ -110,6 +115,17 @@ def test_timetravel() -> None:
     assert list(repo.ancestry(tag="v1.0")) == parents
     assert list(repo.ancestry(branch="feature-not-dead")) == parents
 
+    # check async ancestry works
+    assert list(repo.ancestry(snapshot=feature_snapshot_id)) == asyncio.run(
+        async_ancestry(repo, snapshot=feature_snapshot_id)
+    )
```

Comment:
This is the other point of the PR, we can retrieve ancestry using an async generator and starting on any arbitrary snapshot